spec/models/agents/twitter_favorites_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe Agents::TwitterFavorites do
+
+  before(:example) do
+
+    @opts = {
+      :username => "khanacademy",
+      :number => "10",
+      :history => "100",
+      :expected_update_period_in_days => "2"
+      }
+
+      @mockobject = Agents::TwitterFavorites.new(name: "khanacademy", options: @opts)
+      @mockobject.service = services(:generic)
+      @mockobject.user = users(:bob)
+      @mockobject.save!
+
+  end
+
+  describe "making sure check method works" do
+    before(:example) do
+      stub_request(:any, /khanacademy/).to_return(body: File.read(Rails.root.join("spec/data_fixtures/user_tweets.json")), status: 200)
+    end
+    it "expect change method to change event" do
+      expect { @mockobject.check }.to change {Event.count}.by(4)
```

Comment:
This is great.

I'd also suggest looking at `@agent.events.last` and asserting that its `payload` looks how you expect it to.